kortex_driver/src/hardware_interface.cpp

```diff
@@ -1040,6 +1006,125 @@ void KortexMultiInterfaceHardware::sendTwistCommand()
+          k_api::Base::SafetyIdentifier(feedback_.base().warning_bank_b()));
+        RCLCPP_WARN(LOGGER, "Base returned warning_bank_b: %s", warning_name.c_str());
+      }
+      // check each actuator for faults and warnings and report any values found
+      for (std::size_t i = 0; i < actuator_count_; i++)
+      {
+        if (feedback_.actuators(i).fault_bank_a())
+        {
+          auto fault_name = k_api::Base::SafetyIdentifier_Name(
+            k_api::Base::SafetyIdentifier(feedback_.actuators(i).fault_bank_a()));
+          RCLCPP_ERROR(LOGGER, "Joint[%zu] returned fault_bank_a: %s", i, fault_name.c_str());
+        }
+        if (feedback_.actuators(i).warning_bank_a())
+        {
+          auto warning_name = k_api::Base::SafetyIdentifier_Name(
+            k_api::Base::SafetyIdentifier(feedback_.actuators(i).warning_bank_a()));
+          RCLCPP_WARN(LOGGER, "Joint[%zu] returned warning_bank_a: %s", i, warning_name.c_str());
+        }
+        if (feedback_.actuators(i).fault_bank_b())
+        {
+          auto fault_name = k_api::Base::SafetyIdentifier_Name(
+            k_api::Base::SafetyIdentifier(feedback_.actuators(i).fault_bank_b()));
+          RCLCPP_ERROR(LOGGER, "Joint[%zu] returned fault_bank_b: %s", i, fault_name.c_str());
+        }
+        if (feedback_.actuators(i).warning_bank_b())
+        {
+          auto warning_name = k_api::Base::SafetyIdentifier_Name(
+            k_api::Base::SafetyIdentifier(feedback_.actuators(i).warning_bank_b()));
+          RCLCPP_WARN(LOGGER, "Joint[%zu] returned warning_bank_b: %s", i, warning_name.c_str());
+        }
+      }
+      reset_fault_async_success_ = 0.0;  // tell the fault controller we were unsuccessful
+      reset_fault_cmd_ = NO_CMD;         // clear the reset_fault_cmd
+      return;
+    }
+    // ClearFaults() was successful, switch back to original servoing mode
+    if (
+      arm_mode_ == k_api::Base::ServoingMode::SINGLE_LEVEL_SERVOING ||
+      arm_mode_ == k_api::Base::ServoingMode::LOW_LEVEL_SERVOING)
+    {
+      servoing_mode_hw_.set_servoing_mode(arm_mode_);
+      base_.SetServoingMode(servoing_mode_hw_);
+      std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    }
+    reset_fault_async_success_ = 1.0;  // report success!
+  }
+  catch (k_api::KDetailedException & ex)
+  {
+    RCLCPP_ERROR(LOGGER, "Kortex exception while clearing the faults.");
+    RCLCPP_ERROR(
+      LOGGER, "Error code: %s\n",
+      k_api::ErrorCodes_Name(ex.getErrorInfo().getError().error_code()).c_str());
+    RCLCPP_ERROR(
+      LOGGER, "Error sub code: %s\n",
+      k_api::SubErrorCodes_Name(k_api::SubErrorCodes(ex.getErrorInfo().getError().error_sub_code()))
+        .c_str());
+    RCLCPP_ERROR(LOGGER, "Error description: %s\n", ex.what());
+    reset_fault_async_success_ = 0.0;
+    reset_fault_cmd_ = NO_CMD;  // clear the reset_fault_cmd
+  }
+  catch (std::runtime_error & ex_runtime)
```

Comment:
The previous version had a default catch statement. Is that no longer needed?

Reply:
I could be wrong but I think this was there to catch the other possible exceptions other than `KDetailedException` and `runtime_error`, I think the `base_cyclic_.RefreshFeedback()` might also throw a `std::future_error` and a `std::exception` based on the other try catch blocks in the code.

Reply:
Catching std::exception is always a good idea as it gets you a `what()` to print. Doing `catch(...)` is just sending it down the drain without any signal of what's up, not really good for the long run

Reply:
I updated the code to include the blanket catch with a print that asks the developer to update the code to catch it properly.